Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm

```diff
@@ -1372,6 +1373,56 @@ static bool getEventTypeFromWebEvent(const WebEvent& event, NSEventType& eventTy
     return foundMatch;
 }
 
+WebCore::DictionaryPopupInfo PDFPlugin::dictionaryPopupInfoForSelection(PDFSelection *selection, WebCore::TextIndicatorPresentationTransition presentationTransition)
+{
+    DictionaryPopupInfo dictionaryPopupInfo;
+    if (!selection.string.length)
+        return dictionaryPopupInfo;
+
+    NSAttributedString *nsAttributedString = [selection] {
+        static constexpr unsigned maximumSelectionLength = 250;
+        if (selection.string.length > maximumSelectionLength)
+            return [selection.attributedString attributedSubstringFromRange:NSMakeRange(0, maximumSelectionLength)];
+        return selection.attributedString;
+    }();
+    RetainPtr scaledNSAttributedString = adoptNS([[NSMutableAttributedString alloc] initWithString:[nsAttributedString string]]);
+
+    auto scaleFactor = contentScaleFactor();
+
+    [nsAttributedString enumerateAttributesInRange:NSMakeRange(0, [nsAttributedString length]) options:0 usingBlock:^(NSDictionary *attributes, NSRange range, BOOL *stop) {
+        RetainPtr<NSMutableDictionary> scaledAttributes = adoptNS([attributes mutableCopy]);
+
+        CocoaFont *font = [scaledAttributes objectForKey:NSFontAttributeName];
+        if (font) {
+            auto desiredFontSize = font.pointSize * scaleFactor;
+#if USE(APPKIT)
+            font = [[NSFontManager sharedFontManager] convertFont:font toSize:desiredFontSize];
+#else
+            font = [font fontWithSize:desiredFontSize];
+#endif
```

Comment:
This file is `PLATFORM(MAC)` only, so we don't have to be so considerate! But also, no harm in keeping this around.